compiler/rustc_const_eval/src/interpret/place.rs

```diff
@@ -305,6 +305,18 @@ where
         Ok(mplace)
     }
 
+    #[instrument(skip(self), level = "debug")]
+    pub fn deref_mplace(
```

Comment:
"dereferencing a place" does not make a lot of sense as a concept, and the code doesn't even call `deref_operand`. So something seems odd here.

Is this supposed to be short for something like
```
self.deref_operand(&self.read_immediate(&src.into()).into())
```
? I don't think we need a dedicated function for that, it seems short enough -- and the act of reading the immediate (i.e. the ptr) from the place should rather be left explicit, IMO.

Reply:
Didn't see the `From` implementation, that does make this unnecessary.